documentation/language_server.tex

```diff
@@ -228,34 +226,40 @@ \section{Example: hover request handling}
 
 There are three threads running in the language server:
 \begin{itemize}
-	\item LSP read thread --- a thread in which the \TT{dispatcher} reads messages from a standard input.
+	\item LSP read thread --- a thread in which the \TT{dispatcher} reads messages from the standard input.
 	\item DAP read thread --- a thread in which the \TT{tcp\_handler} listens on a localhost port to initiate a DAP session. After accepting the DAP client, the \TT{dispatcher} reads DAP input on this thread too.
 	\item Worker thread in \TT{request\_manager} that processes each request using the \TT{lsp\_server} or the \TT{dap\_server} and ultimately the parser library.
 \end{itemize}
 
-The threads are synchronized in two ways: firstly, there is a mutex that protects adding to the request queue simultaneously by the LSP and the DAP read threads. Secondly, there is a conditional variable to control the worker thread so it sleeps when there are no requests and wakes up when a new request has been added.
+The threads are synchronized in two ways: First, there is a mutex that protects adding to the request queue simultaneously by the LSP and the DAP read threads. Second, there is a conditional variable to control the worker thread.
 
-In the request manager, there is a mechanism for invalidating requests that have been obsoleted by new requests. When a new request comes, all previous requests (including the currently processed one) that are concerning the same file are invalidated, but they cannot be simply removed from the queue and they still have to be processed. However, the parser library gets the information that the request has been obsoleted by a newer one and may behave differently.
+Request manager additionally incorporates a mechanism for invalidating requests that have been obsoleted by new requests. The obsoleting of requests is done by a cancellation token. It is shared between the parser library and the \TT{request\_manager}. When set to true, the results of current request or notification are no longer needed, the parser library stops all parsing and return as soon as possible.
 
-For example, when a user starts changing a file, every character he writes is passed to the language server as a textDocument/didChange notification. Normally, each such notification is processed in two stages:
+When a new request arrives, all previous requests (including the currently processed one) that concern the same file are invalidated, but they cannot be simply removed from the queue. They still have to be processed, because they may carry information that must not be discarded. The parser library processes the request but does not reparse any source files.
```

Comment:
```suggestion
When a new request arrives, all previous requests (including the currently processed one) that concern the same file are invalidated. However, they cannot be simply removed from the queue. They still have to be processed as they may carry information that must not be discarded. The parser library processes the request but does not reparse any source files.
```

Reply:
ako citatel som po precitani odstavca zvedavy aku informaciu carryuju ze ich nemozno discartnut. Je to niekde spomenute? alebo je to kontextualne jasne?

Reply:
je to jasne az z toho prikladu co pojde potom.. mozem dat nieco kratke do zatvorky